esp/src/src/ESPRequest.ts

```diff
@@ -84,6 +85,15 @@ class RequestHelper {
         }
 
         return retVal.then(function (response) {
+            if (lang.exists("Exceptions.Exception", response)) {
```

Comment:
Is this safe - we receive exceptions for other reasons?

Reply:
What do you think about `if (lang.exists("Exceptions.Exception.Code === "401"))`

Reply:
I added an extra layer to check for the actual Exception code which is  "401"